adaptor-rm-webapp/src/main/java/com/sample/rm/StorePool.java

```diff
@@ -0,0 +1,128 @@
+
+    // Start of user code class_methods
+    // End of user code
+
+    public StorePool (URI sparqlQueryUrl, URI sparqlUpdateUrl, URI defaultNamedGraphUri, String userName, String password) throws StoreAccessException {
+        this.defaultNamedGraphUri = defaultNamedGraphUri;
+        this.storePool = new ArrayList<>(INITIAL_POOL_SIZE);
+        this.usedStores = new ArrayList<>();
+        // Start of user code constructor_init_1
+        // End of user code
+        for (int i = 0; i < INITIAL_POOL_SIZE; i++) {
+            Store s = null;
+            if( userName != null && password != null ){
+                s = StoreFactory.sparql(sparqlQueryUrl.toString(), sparqlUpdateUrl.toString(), userName, password);
+            }else{
+                s = StoreFactory.sparql(sparqlQueryUrl.toString(), sparqlUpdateUrl.toString());
+            }
+            storePool.add(s);
+        }
+        // Start of user code constructor_finalize_1
+        // End of user code
+    }
+
+    public static StorePool create (String propertiesFileName) throws StoreAccessException {
+        // Start of user code create_init
+        // End of user code
+        Properties lyoStoreProperties = new Properties();
+        String lyoStorePropertiesFile = StorePool.class.getResource(propertiesFileName).getFile();
+        try {
+            lyoStoreProperties.load(new FileInputStream(lyoStorePropertiesFile));
+        } catch (IOException e) {
+            log.error("Failed to initialize Store. properties file for Store configuration could not be loaded.", e);
+            throw new StoreAccessException(e);
+        }
+
+        URI defaultNamedGraph;
+        URI sparqlQueryUrl;
+        URI sparqlUpdateUrl;
+        try {
+            defaultNamedGraph = new URI(lyoStoreProperties.getProperty("defaultNamedGraph"));
+            sparqlQueryUrl = new URI(lyoStoreProperties.getProperty("sparqlQueryUrl"));
+            sparqlUpdateUrl = new URI(lyoStoreProperties.getProperty("sparqlUpdateUrl"));
+        } catch (URISyntaxException e) {
+            log.error("Failed to initialize Store. One of the configuration property ('defaultNamedGraph' or 'sparqlQueryUrl' or 'sparqlUpdateUrl') is not a valid URI.", e);
+            throw new StoreAccessException(e);
+        }
+        String userName = lyoStoreProperties.getProperty("username");
+        String password = lyoStoreProperties.getProperty("password");
+        // Start of user code create_finalize
+        // End of user code
+        return new StorePool(sparqlQueryUrl, sparqlUpdateUrl, defaultNamedGraph, userName, password);
+    }
+
+    public URI getDefaultNamedGraphUri() {
+        return defaultNamedGraphUri;
+    }
+
+    public Store getStore() {
+        if (storePool.isEmpty()) {
+            throw new RuntimeException("Maximum pool size reached, no available store connections!");
+        }
+        Store s;
+        synchronized(lock) {
+            s = storePool.remove(storePool.size() - 1);
```

Comment:
you need to check `storePool.isEmpty` here as well.

Reply:
True! Shall fix.